README.md

```diff
@@ -153,6 +156,106 @@ keys used to populate the parameters of the S3 table function:
 
 See the [S3 test file](https://github.com/ClickHouse/dbt-clickhouse/blob/main/tests/integration/adapter/test_s3.py) for examples of how to use this macro.
 
+# Distributed materializations
+
+WARNING: 
+
+To use distributed materializations correctly you should set **insert_distributed_sync** = 1 or use as prehhook in order to have correct data while SELECT queries. Otherwise downstream calculation could be wrong if the distributed insert is not finished in time
+
+## Distributed table materialization
+
+Distributed table created with following steps:
+1. Creates temp view with sql query to get right structure
+2. Create empty local tables based on view 
+3. Create distributed table based on local tables. 
+4. Data inserts into distributed table, so it is distributed across shards without duplicating.
+
+### Distributed table model example
+```sql
+{{
+    config(
+        materialized='distributed_table',
+        on_cluster='cluster',
+        order_by='id, created_at',
+        sharding_key='cityHash64(id)',
+        engine='ReplacingMergeTree'
+    )
+}}
+
+select id, created_at, item from {{ source('db', 'table') }}
+```
+
+### Generated migrations
+
+```sql
+CREATE TABLE db.table_local on cluster cluster
+(
+    `id` UInt64,
+    `created_at` DateTime,
+    `item` String
+)
+ENGINE = ReplacingMergeTree
+ORDER BY (id, created_at)
+SETTINGS index_granularity = 8192;
+
+
+CREATE TABLE db.table on cluster cluster
+(
+    `id` UInt64,
+    `created_at` DateTime,
+    `item` String
+)
+ENGINE = Distributed('cluster', 'db', 'table_local', cityHash64(id));
+```
+
+## Distributed incremental materialization
+
+Incremental model based on the same idea as distributed table, the main difficulty is to process all incremental strategies correctly.
+
+1. _The Append Strategy_ just insert data into distributed table.
+2. _The Delete+Insert_ Strategy creates distributed temp table to work with all data on every shard.
+3. _The Default (Legacy) Strategy_ creates distributed temp and intermediate tables for the same reason.
+
+Only shard tables are replacing, because distributed table does not keep data. 
+The distributed table reloads only when the full_refresh mode is enabled or the table structure may have changed.
+
+### Distributed incremental model example
+```sql
+{{
+    config(
+        materialized='distributed_incremental',
+        on_cluster='cluster',
```

Comment:
Should it just use the `on_cluster_clause` macro instead of specifying it as a parameter?  I think every other operation will require that macro to be automatically applied using the "active" cluster, so if the wrong cluster is specified here, that would break stuff?

Reply:
Actually it looks like that's what you do, so we just need to remove it from the example here?

Reply:
Yes, removed